src/Colors/index.js

```diff
@@ -0,0 +1,101 @@
+import { mergeColors, calculateAlpha, isObject, isString } from './utils.js'
+
+let colors = {
+  white: '#ffffff',
+  black: '#000000',
+  red: '#ff0000',
+  green: '#00ff00',
+  blue: '#0000ff',
+  yellow: '#feff00',
+  cyan: '#00feff',
+  magenta: '#ff00ff',
+}
+
+const normalizedColors = {
+  //store for normalized colors
+}
+
+const calculateColor = (value, options) => {
+  let targetColor = normalizeColor(value)
+  if (!isNaN(options)) {
+    options = { alpha: options }
+  }
+  if (options && isObject(options)) {
+    if (!isNaN(options.darken)) {
+      targetColor = mergeColors(targetColor, 0xff000000, 1 - options.darken)
```

Comment:
maybe we should check if options.darken is a number between 0 and 1?

Reply:
made a function for limiting the number within the range of 0 - 1